pkg/alertmanager/amcfg.go

```
@@ -1603,6 +1601,37 @@ func (cb *configBuilder) convertTLSConfig(in *monitoringv1.SafeTLSConfig, crKey
 	return &out
 }
 
+func (cb *configBuilder) convertProxyConfig(in monitoringv1.ProxyConfig, crKey types.NamespacedName) proxyConfig {
+	out := proxyConfig{}
+
+	if in.ProxyURL != nil {
+		out.ProxyURL = *in.ProxyURL
+	}
+
+	if in.NoProxy != nil {
+		out.NoProxy = *in.NoProxy
+	}
+
+	if in.ProxyFromEnvironment != nil {
+		out.ProxyFromEnvironment = *in.ProxyFromEnvironment
+	}
+
+	if len(in.ProxyConnectHeader) > 0 {
+		proxyConnectHeader := make(map[string][]string, len(in.ProxyConnectHeader))
+		s := cb.store.ForNamespace(crKey.Namespace)
+		for k, v := range in.ProxyConnectHeader {
+			proxyConnectHeader[k] = []string{}
+			for _, vv := range v {
+				value, _ := s.GetSecretKey(vv)
```

we need to return the error if not nil.

---

Thanks. Add the error return